README.developers.md

```diff
@@ -1,51 +1,159 @@
 # Commit Procedures
 
-## For external developers
-See [Submitting Code to VTR](CONTRIBUTING.md#submitting-code-to-vtr).
+For general guidance on contributing to VTR see [Submitting Code to VTR](CONTRIBUTING.md#submitting-code-to-vtr).
 
-## For developers with commit rights
-The guiding principle in internal development is to submit your work into the repository without breaking other people's work.
-When you commit, make sure that the repository compiles, that the flow runs, and that you did not clobber someone else's work.
-In the event that you are responsible for "breaking the build", fix the build at top priority.
+The actual machanics of submitting code are outlined below.
 
-We have some guidelines in place to help catch most of these problems:
+However they differ slightly depending on whether you are:
+ * an **internal developer** (i.e. you have commit access to the main VTR repository at `github.com/verilog-to-routing/vtr-verilog-to-routing`) or, 
+ * you an (**external developer**) (i.e. no commit access).
 
-1.  Before you push code to the central repository, your code MUST pass the check-in regression test.
-    The check-in regression test is a quick way to test if any part of the VTR flow is broken.
+The overall approach is similar, but we call out the differences below.
 
-    At a minimum you must run:
-    ```shell
-    #From the VTR root directory
-    $ ./run_reg_test.pl vtr_reg_basic
-    ```
-    You may push if all the tests return `All tests passed`.
+1. Setup a local repository on your development machine.
 
-    However you are strongly encouraged to run both the *basic* and *strong* regression tests:
-    ```shell
-    #From the VTR root directory
-    $ ./run_reg_test.pl vtr_reg_basic vtr_reg_strong
+    a. **External Developers**
+
+    * Create a 'fork' of the VTR repository.
+
+        Usually this is done on GitHub, giving you a copy of the VTR repository (i.e. `github.com/<username>/vtr-verilog-to-routing`, where `<username>` is your GitHub username) to which you have commit rights.
+        See [About forks](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/about-forks) in the GitHub documentation.
+
+    * Clone your 'fork' onto your local machine.
+    
+        For example, `git clone git@github.com:<username>/vtr-verilog-to-routing.git`, where `<username>` is your GitHub username.
+
+    b. **Internal Developers**
+
+    * Clone the main VTR repository onto your local machine.
+    
+        For example, `git clone git@github.com:verilog-to-routing/vtr-verilog-to-routing.git`.
+
+2. Move into the cloned repository.
+
+    For example, `cd vtr-verilog-to-routing`.
+
+3. Create a *branch*, based off of `master` to work on.
+
+    For example, `git checkout -b my_awsome_branch master`, where `my_awsome_branch` is some helpful and descriptive name you give you're branch.
```

Comment:
awsome --> awesome (unless the spelling is intentional!)

Reply:
Could delete *good* or put descriptive in brackets (reads a bit strangely right now)